waku/node/validator_signed.nim

```diff
@@ -19,9 +19,9 @@ import
 const MessageWindowInSec = 5*60 # +- 5 minutes
 
 import
```

Comment:
unsure if @jm-clius agrees.

Reply:
Better having the signed validator in `apps`? the idea is to have all the pieces to build a Waku node with standard features under `waku/node` (but without adding it to the `WakuNode` object itself).

Thought that a signed validator entered into the set of "standard features"

Reply:
However, these elements (app-level validation, app's DB config, etc.) were moved out of the node for a reason. This (incorrectly) brings back awareness of the app environment into the node object, while the app environment should know about the node and not the other way around. Amongst other things, you're now seeing a lot of interesting circular imports (such as `internal_config` depending on `waku_node`, while `waku_node` also depends on `internal_config`).

Perhaps the solution is rather to raise the bindings to the app level and, of course, create a convenient app-level node factory or middleware.

Reply:
> the solution is rather to raise the bindings to the app level and, of course, create a convenient app-level node factory or middleware.

:+1: !

Reply:
Makes sense, maybe all of this should be at app level.

Notice that here we did not change the node object at all - we only moved the procs that initialize nodes from the `app` file to a `node_factory` file inside `waku/node` directory which isn't used by any other file related to the node.

Or in other words, we're attaching to `waku_node` a separate "kit" to easily create a node. So once you import `waku_node` you get both the basic procs to do everything from scratch and a separate already built "node factory" with standard implementations.

Regarding the circular imports, notice that those two are different files. `internal_config` depends on `waku/node/waku_node.nim` which has the actual `WakuNode` implementation. And then there's `waku/waku_node.nim` that imports files from the `waku/node` directory and exports them as a bundle so apps don't have to import all the files individually.
Here we're adding the node factory to the second one, to `waku/waku_node.nim`. But we are not touching the node's implementation.

Maybe we should have all of this at the app level. But the only thing that would change here is the file path, the logic is already separate.

Reply:
Right, good point about the import here - `waku/waku_node.nim` is just a barrel import. However, it remains a bit weird that there are now "app" imports in modules that live very closely with the node. It indeed only affects import and directory structure afaics, but I guess what we need is:
a) node (the wrapper around libp2p switch)
b) app middleware (config, factory, validators, etc. that's used by all apps, including ones using the bindings)
c) specific apps, e.g. wakunode2 - bare minimum to use the app middleware to create a specific type of waku application, such as `wakunode2` binary.

This layered approach should also reflect in the import structure (outer app can import app middleware can import node, but never the other way).

Reply:
Sounds great! Really like that idea of a new "middleware" layer - agree 100% that it's a much more logically sound structure.

Will re-organize the it like that.
Thanks so much! 🤩